identity-wallet/src/state/credentials/reducers/handle_oid4vp_authorization_request.rs

```diff
@@ -124,44 +123,34 @@ pub async fn handle_oid4vp_authorization_request(mut state: AppState, action: Ac
         }
         info!("response successfully sent");
 
-        let connection_time = chrono::Utc::now().to_rfc3339();
-
         let (client_name, _logo_uri, connection_url) =
             get_oid4vp_client_name_and_logo_uri(&oid4vp_authorization_request)
                 .map_err(|_| MissingAuthorizationRequestParameterError("connection_url"))?;
 
-        let mut connections = state.connections.clone();
-        let result = connections
-            .iter_mut()
-            .find(|connection| connection.url == connection_url && connection.client_name == client_name)
-            .map(|connection| {
-                connection.last_interacted = connection_time.clone();
-            });
-
-        let connection_id = Connection::create_connection_id(&client_name);
-
-        if result.is_none() {
-            connections.push(Connection {
-                id: connection_id.to_string(),
-                client_name: client_name.to_string(),
-                url: connection_url,
-                verified: false,
-                first_interacted: connection_time.clone(),
-                last_interacted: connection_time.clone(),
-            })
-        };
-
-        state.connections = connections;
+        let previously_connected = state.connections.contains(connection_url.as_str(), &client_name);
+        let mut connections = state.connections;
+        let connection = connections.insert_or_update(&connection_url, &client_name);
 
         // History
+        if !previously_connected {
+            // Only add a `ConnectionAdded` event if the connection was not previously connected.
+            state.history.push(HistoryEvent {
```

Comment:
I had a similar idea of abstracting the `HistoryEvents` that I added to this PR description. We can definitely do something like you suggest although I think it's slightly out of the scope of this PR

Reply:
You mean that there would be no more Vec<HistoryEvent> field in the Appstate Joris? that this would be become a substruct of sorts to the Connection struct?